driver/src/main/java/org/neo4j/driver/v1/Value.java

```diff
@@ -194,12 +195,36 @@
      */
     Object asObject();
 
+    /**
+     * Apply the mapping function on the value if the value is not a {@link NullValue}, or the default value if the value is a {@link NullValue}.
+     * @param mapper The mapping function defines how to map a {@link Value} to T.
+     * @param defaultValue the value to return if the value is a {@link NullValue}
+     * @param <T> The return type
+     * @return The value after applying the given mapping function or the default value if the value is {@link NullValue}.
+     */
+    <T>T computeOrDefault( Function<Value, T> mapper, T defaultValue );
+
+    /**
+     * Apply the mapping function on the value if the value is not a {@link NullValue}, or null if the value is a {@link NullValue}.
+     * @param mapper The mapping function defines how to map a {@link Value} to T.
+     * @param <T> The return type
+     * @return The value after applying the given mapping function or null if the value is {@link NullValue}.
+     */
+    <T>T computeOrNull( Function<Value, T> mapper );
+
     /**
      * @return the value as a Java boolean, if possible.
      * @throws Uncoercible if value types are incompatible.
      */
     boolean asBoolean();
 
+    /**
+     * @param defaultValue return this value if the value is a {@link NullValue}.
```

Comment:
Maybe add description and `@throws Uncoercible if value types are incompatible.`

Reply:
It was in the javadoc.